Formula/g/gcc.rb

```diff
@@ -120,24 +120,22 @@ def install
       # "Updated load commands do not fit in the header"
       make_args = %w[BOOT_LDFLAGS=-Wl,-headerpad_max_install_names]
     else
-      # Fix cc1: error while loading shared libraries: libisl.so.15
-      args << "--with-boot-ldflags=-static-libstdc++ -static-libgcc #{ENV.ldflags}"
-
       # Fix Linux error: gnu/stubs-32.h: No such file or directory.
       args << "--disable-multilib"
 
       # Enable to PIE by default to match what the host GCC uses
       args << "--enable-default-pie"
 
+      # Use GNU hash tables similar to Linux distros for better performance
+      args << "--with-linker-hash-style=gnu"
```

Comment:
Trying this out too. Didn't backport to versioned GCC as not sure of compatibility impact.

I think most Linux distros already do this or something similar.
- Arch Linux[^1]
- Fedora[^2]
- Debian[^3] - via patch

[^1]: https://gitlab.archlinux.org/archlinux/packaging/packages/gcc/-/blob/main/PKGBUILD?ref_type=heads#L84
[^2]: https://src.fedoraproject.org/rpms/gcc/blob/rawhide/f/gcc.spec#_1177
[^3]: https://salsa.debian.org/toolchain-team/gcc/-/blob/master/debian/patches/gcc-hash-style-gnu.diff?ref_type=heads